components/incidents/incidents-list-view/loadPageData.tsx

```diff
@@ -0,0 +1,134 @@
+import { useEffect, useState } from "react";
+import { NEXT_PUBLIC_AUTH_TOKEN } from "../../../constants";
+import { PageType } from "./incident_list_view";
+
+/**
+ * Loads data from API
+ * @param pageNumber pageNumber required for paginated data
+ * @param pageType type of page
+ * @returns API response and API loading status
+ */
+
+const limit = 10; // determines the no. of rows of data in a page
+
+export default function LoadPageData(pageNumber: number, pageType: string) {
+  const [dataList, setData] = useState<any[]>(Array()); // stores data response from API
+  const [hasLoaded, setHasLoaded] = useState<boolean>(false); // status of loading data from API
+  const [hasMore, setHasMore] = useState<boolean>(true); // do we have to fetch more data
+
+  let idList: string[] | undefined = []; // list of pageIDs from which we will be fetching data from
+
+  /**
+   * Get the ID list from API and then call for data in each page
+   * @param pageNumber pageNumber for getting paginated data
+   * @param pageType type of page to be rendered
+   */
+  const getIDData = async (pageNumber: number, pageType: string) => {
+    try {
+      const URL = "https://api.statuspage.io/v1/pages";
+      const response = await fetch(URL, {
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `OAuth ${NEXT_PUBLIC_AUTH_TOKEN ?? ""}`,
+        },
+      });
+      const myJson = await response.json();
+      idList = myJson.map((data: any) => data["id"]);
+      // console.log(idList);
+    } catch (err) {
+      alert("Too many API calls");
+      console.log(err);
+    }
+
+    // call getData for each pageID
+    if (idList !== undefined) {
+      for (let i = 0; i < idList.length; i++) {
+        getData(idList[i], pageNumber, pageType);
```

Comment:
what is idList ?

Reply:
The list of pageIDs from which we need to fetch incident-list-data. Renamed it to pageIDsList.